### PR TITLE
Add span events to New Relic exporter

### PR DESCRIPTION
**Pull request by justinfoote**
**Description:**
Adding a Feature: SpanEvent support in the New Relic exporter.

**Testing:** 
I've added unit tests that verify the struct is constructed the way we expect when given an OpenTelemetry Span that contains events. 
The underlying serialization is done by the New Relic Telemetry SDK, which has its own automated tests to verify the serialization.
I've also used a locally-build collector to verify that these structs are serialized and sent to New Relic correctly, and that the format matches our expectations. 
The output from the collector with debug logging enabled:
```
2020-11-19T17:10:07.482Z	info	newrelicexporter/newrelic.go:47	2020/11/19 17:10:07.457528 {"body-length":286,"event":"data post","url":"https://staging-trace-api.newrelic.com/trace/v1"}
	{"component_kind": "exporter", "component_type": "newrelic", "component_name": "newrelic"}
2020-11-19T17:10:07.563Z	debug	newrelicexporter/newrelic.go:47	2020/11/19 17:10:07.563437 {"data":[{"common":{},"spans":[{"id":"3e0f5885710776cd","trace.id":"3e0f5885710776cd3e0f5885710776cd","timestamp":1605748201371,"attributes":{"name":"post","duration.ms":508.068,"collector.name":"opentelemetry-collector","collector.version":"0.0.0","span.kind":"client","service.name":"service-1","net.host.ip":"127.0.0.1","net.host.port":8080},"events":[{"name":"cs","timestamp":1605748201371,"attributes":{}},{"name":"cr","timestamp":1605748201371,"attributes":{}}]},{"id":"3e0f5885710776de","trace.id":"3e0f5885710776cd3e0f5885710776cd","timestamp":1605748201371,"attributes":{"name":"service 2 span","parent.id":"3e0f5885710776cd","duration.ms":2.019,"net.host.ip":"127.0.0.1","net.host.port":8080,"collector.name":"opentelemetry-collector","collector.version":"0.0.0","span.kind":"client","service.name":"service-2"}}]}],"event":"uncompressed request body","url":"https://staging-trace-api.newrelic.com/trace/v1"}
	{"component_kind": "exporter", "component_type": "newrelic", "component_name": "newrelic"}
2020-11-19T17:10:09.217Z	info	newrelicexporter/newrelic.go:47	2020/11/19 17:10:09.217161 {"body":{"requestId":"f1f565fb-0001-b000-0000-0175e17c1e10"},"event":"data post response","status":202}
```
This format matches our ingest expectations.

### CODE REVIEW

File: exporter/newrelicexporter/transformer.go [hunk 1]
```diff
@@ -160,6 +161,26 @@ func (t *traceTransformer) SpanAttributes(span pdata.Span) map[string]interface{
 	return attrs
 }
 
+func (t *traceTransformer) SpanEvents(span pdata.Span) []telemetry.Event {
```
Comment (MrAlias) [inline exporter/newrelicexporter/transformer.go:164]: Missing documentation

```suggestion
// SpanEvents transforms the recorded events of span into New Relic tracing events.
func (t *traceTransformer) SpanEvents(span pdata.Span) []telemetry.Event {
```

File: exporter/newrelicexporter/transformer.go [hunk 2]
```diff
@@ -160,6 +161,26 @@ func (t *traceTransformer) SpanAttributes(span pdata.Span) map[string]interface{
...
+	length := span.Events().Len()
+	if length == 0 {
+		return nil
+	}
+
+	events := make([]telemetry.Event, length)
+
+	for i := 0; i < span.Events().Len(); i++ {
```
Comment (MrAlias) [inline exporter/newrelicexporter/transformer.go:172]:
```suggestion
	for i := 0; i < length; i++ {
```

File: exporter/newrelicexporter/transformer.go [hunk 3]
```diff
@@ -160,6 +161,26 @@ func (t *traceTransformer) SpanAttributes(span pdata.Span) map[string]interface{
...
+		event := span.Events().At(i)
+		evt := telemetry.Event{
+			EventType:  event.Name(),
+			Timestamp:  pdata.UnixNanoToTime(event.Timestamp()),
+			Attributes: tracetranslator.AttributeMapToMap(event.Attributes()),
+		}
+		events[i] = evt
```
Comment (MrAlias) [inline exporter/newrelicexporter/transformer.go:179]: nit

```suggestion
		events[i] = telemetry.Event{
			EventType:  event.Name(),
			Timestamp:  pdata.UnixNanoToTime(event.Timestamp()),
			Attributes: tracetranslator.AttributeMapToMap(event.Attributes()),
		}
```

**Review by MrAlias — APPROVED** [review]

File: exporter/newrelicexporter/transformer.go (hunk 2 above)
Comment (justinfoote) [reply exporter/newrelicexporter/transformer.go:172]: :man_facepalming:

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**